### ISSUE TITLE
Add slf4j implementation to test classpath of all modules that use logging

### ISSUE THREAD

**Issue by robobario** [kroxylicious/kroxylicious#1676]
Some modules that use slf4j apis for logging do not have an slf4j backend on the classpath at unit test time. This means to debug unit tests we have to add something like log4j to the pom.xml and add a configuration file. All a bit repetitive to figure out what is missing per module to get some log output.

Each module with slf4j being used for logging should have a logging implementation available at testing time, logging can be configured off by default but it would be nice if enabling logging for tests was a matter of uncommenting a line in the config file or setting a system property (something along those lines).

**Comment by SamBarker** [comment]
Do we need anything other than `slf4j-simple` on the test class path?

**Comment by robobario** [comment]
Calling out unit testing wasn't quite right, we have a lot of tests sitting somewhere on this unit -> integration spectrum that aren't named IT.

The example spurring it was ec2 metadata integration, which is more in that integration area with async code and a mock http server. I can't remember what I was poking with, but I wanted logs on. Once async code is involved the logging becomes a more useful tool.

It's more that logging is a facility that may be useful and there should be a consistent way to access it. For that reason I'd prefer log4j so we match other modules test setup.

**Comment by tombentley** [comment]
I think the relevance of this has changed in light of things like Claude code needing to fiddle with logging (in the absence of it being able to use a debugger) in order to do debugging. 

I agree with what @robobario seems to be saying: 

* Getting the right modules on the classpath in tests it more work that it should be.
* It's better to have a consistent way to do it, than the confusion that results from some modules using `slf4j-simple` others doing log4j2.

**Comment by robobario** [comment]
Yep that was what I was trying to say. There's no consistency. I want to look for the test logging config in a standard location and have it look familiar for every module. 

As opposed to figuring out if there is an impl available, if so which kind, if so does it have a config file etc.
